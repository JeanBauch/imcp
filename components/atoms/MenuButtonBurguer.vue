<template>
	<div class="menu-btn" @click="handleClickOpenMenu()">
		<div class="menu-burguer" />
	</div>
</template>

<script lang="ts">
import Vue from 'vue';
import { toggleClass } from '~/utils';

export default Vue.extend({
	data () {
		return {
			menuOPen: false,
		};
	},
	methods: {
		handleClickOpenMenu () {
			const menuBtn = document.querySelector('.menu-btn');
			const navList = document.getElementById('navList');
			if (!this.menuOPen) {
				toggleClass(menuBtn, 'open');
				toggleClass(navList, 'open');
				this.menuOPen = true;
			} else {
				toggleClass(menuBtn, 'open');
				toggleClass(navList, 'open');
				this.menuOPen = false;
			}
		},
	},
});
</script>
