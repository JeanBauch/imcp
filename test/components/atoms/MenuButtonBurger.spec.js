import { shallowMount } from '@vue/test-utils';

import MenuButtonBurguer from '~/components/atoms/MenuButtonBurguer.vue';
// import NavBarMobile from '~/components/molecules/NavbarMobile.vue';

describe('Components:atoms:MenuButtonBurguer', () => {
	let mountConfig;

	beforeEach(() => {
		mountConfig = {
			data () {
				return {
					menuOPen: false,
				};
			},
		}; // Passar aqui o localVue para VueStore e as propsData caso precise
	});

	const buildWrapper = () => {
		// const store = configureStore(); *Caso tenha uma Store futura, preparar os states
		return {
			...mountConfig, // Caso tenha uma store, retornar aqui como \store,\
		};
	};

	describe('Snapshots', () => {
		it('should render component', () => {
			const wrapper = shallowMount(MenuButtonBurguer, {});

			expect(wrapper.html()).toMatchSnapshot();
		});
	});

	describe('Methods', () => {
		it('should create componente with open closed  == { menuOpen:False }', async () => {
			const config = buildWrapper();
			const wrapper = shallowMount(MenuButtonBurguer, config);

			const button = wrapper.find('.menu-btn');

			const handleClickOpenMenu = jest.spyOn(wrapper.vm, 'handleClickOpenMenu');
			await button.trigger('click');

			expect(handleClickOpenMenu).toHaveBeenCalled();
			jest.resetAllMocks();
		});

		it('should create componente with open menu == { menuOpen:True }', async () => {
			const config = buildWrapper();
			const wrapper = shallowMount(MenuButtonBurguer, config);

			await wrapper.setData({ menuOPen: true });

			const button = wrapper.find('.menu-btn');

			const handleClickOpenMenu = jest.spyOn(wrapper.vm, 'handleClickOpenMenu');
			await button.trigger('click');

			expect(handleClickOpenMenu).toHaveBeenCalled();
			jest.resetAllMocks();
		});
	});
});
