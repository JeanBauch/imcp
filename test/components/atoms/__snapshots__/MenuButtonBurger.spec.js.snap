// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:MenuButtonBurguer Snapshots should render component 1`] = `
"<div class=\\"menu-btn\\">
  <div class=\\"menu-burguer\\"></div>
</div>"
`;
